craft_application/git/_git_repo.py

```diff
@@ -193,6 +231,65 @@ def _init_repo(self) -> None:
                 f"Could not initialize a git repository in {str(self.path)!r}."
             ) from error
 
+    def remote_exist(self, remote_name: str) -> bool:
+        """Check if remote with given name is configured locally.
+
+        :param remote_name: the remote repository name
+        """
+        try:
+            _ = self._repo.remotes[remote_name]
+        except KeyError:
+            return False
+        else:
+            return True
+
+    def add_remote(
+        self,
+        remote_name: str,
+        remote_url: str,
+    ) -> None:
+        """Add remote in the repository configuration.
+
+        :param remote_name: the remote repository name
+        :param remote_url: the remote repository URL
+
+        :raises GitError: if remote cannot be created
+        """
+        try:
+            self._repo.remotes.create(remote_name, remote_url)
+        except ValueError as ve:
+            raise GitError(f"remote `{remote_name}` already exists.") from ve
+        except pygit2.GitError as error:
+            raise GitError(
+                f"could not add remote to a git repository in {str(self.path)!r}."
+            ) from error
+
+    def rename_remote(
+        self,
+        remote_name: str,
+        new_remote_name: str,
+    ) -> None:
+        """Rename remote in the repository configuration.
+
+        :param remote_name: the remote repository name
+        :param new_remote_name: the new name for the remote
+
+        :raises GitError: if remote cannot be renamed
+        """
+        logger.debug("Renaming `%s` to `%s`", remote_name, new_remote_name)
+        try:
+            self._repo.remotes.rename(remote_name, new_remote_name)
+        except KeyError as ke:
```

Comment:
Please use normal convention (either `exc` or `error`) for these names.

Reply:
Changed in https://github.com/canonical/craft-application/pull/396/commits/d2f2f7796da819576a9468bc2fa8a6b4612da59a